src/engine/cancun.md

```diff
@@ -122,9 +122,11 @@ Refer to the response for [`engine_forkchoiceUpdatedV2`](./shanghai.md#engine_fo
 
 This method follows the same specification as [`engine_forkchoiceUpdatedV2`](./shanghai.md#engine_forkchoiceupdatedv2) with addition of the following:
 
-1. Client software **MUST** check that provided set of parameters and their fields strictly matches the expected one and return `-32602: Invalid params` error if this check fails. Any field having `null` value **MUST** be considered as not provided.
+1. Client software **MUST** validate and apply the forkchoice state before validating `payloadAttributes`. If `forkchoiceState` does not match the expected fields, clients must return `-32602: Invalid params`.
```

Comment:
```suggestion
1. Client software **MUST** validate and apply the forkchoice state before starting to validate and process `payloadAttributes`. If `forkchoiceState` does not match the expected fields, clients must return `-32602: Invalid params`.
```
I think we should be more specific here e.g. `payloadAttributes` must not even be validated in the case when `SYNCING` is returned by EL. Alternatively, we can make a change to the Paris spec to separate `payloadAttributes` processing from the main flow like encapsulating its validation and the build process  setup in a dedicated step and explicitly saying that this step is only executed when `forkchoiceState` is successfully applied.

This would allow for extending `payloadAttributes` and `forkchoiceState` processing steps in the future without the need to be specific about the relations of those steps. Thoughts on that?

Reply:
Agree with including a caveat for `SYNCING`. `payloadAttributes` validation during `SYNCING` is currently required by the hive test `Invalid PayloadAttributes: Missing BeaconRoot (Syncing) (Cancun)`, which I believe is incorrect.

`payloadAttributes` to build a block with are irrelevant when `SYNCING` and should be ignored instead, since this suggests the CL doesn't know the EL is syncing yet, and should not have requested a new block to be built.